src/lib.rs

```diff
@@ -311,3 +346,82 @@ impl Write for Device {
         Ok(())
     }
 }
+
+impl Device {
+    pub fn set_mpsse_clock(&mut self, freq: u32) -> std::result::Result<(), io::Error> {
+        const MAX: u32 = 30_000_000;
+        const MIN: u32 = 92;
+
+        assert!(
+            freq >= MIN,
+            "frequency of {} exceeds minimum of {}",
+            freq,
+            MIN
+        );
+        assert!(
+            freq <= MAX,
+            "frequency of {} exceeds maximum of {}",
+            freq,
+            MAX
+        );
+
+        let (divisor, clkdiv) = if freq <= 6_000_000 {
+            (6_000_000 / freq - 1, Some(true))
+        } else {
+            (30_000_000 / freq - 1, Some(false))
+        };
+
+        self.write_all(MpsseCmdBuilder::new().set_clock(divisor, clkdiv).as_slice())?;
+
+        Ok(())
+    }
+}
+
+impl MpsseCmdExecutor for Device {
+    type Error = io::Error;
+
+    fn init(&mut self, settings: &MpsseSettings) -> std::result::Result<(), io::Error> {
+        let millis = u8::try_from(settings.latency_timer.as_millis())
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+
+        if settings.reset {
+            self.usb_reset()
+                .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+        }
+
+        self.usb_purge_buffers()
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+
+        self.set_write_chunksize(settings.in_transfer_size);
+        self.set_read_chunksize(settings.in_transfer_size);
+        self.set_latency_timer(millis)
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+        self.usb_set_event_char(0, false)
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+        self.usb_set_error_char(0, false)
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+        self.set_flow_control(FlowControl::RtsCts)
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+        self.set_bitmode(0, BitMode::Reset)
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+        self.set_bitmode(settings.mask, BitMode::Mpsse)
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+
+        if let Some(frequency) = settings.clock_frequency {
+            self.set_mpsse_clock(frequency)
+                .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+        }
+
+        Ok(())
+    }
+
+    fn send(&mut self, data: &[u8]) -> std::result::Result<(), io::Error> {
+        self.usb_purge_tx_buffer()
```

Comment:
Please add a comment documenting why this purge is required.

Reply:
Done